# 에러 처리

에러가 발생하지 않는 코드를 작성하는 것은 불가능하다.
에러가 나면 코드의 실행이 멈추기 때문에 적절한 에러처리로 방지해야 한다.

## try...catch...finally문

try 실행할 코드(에러 발생 가능성)
catch 에러가 발생하면 이 블록에서 실행
finally 에러 발생과 상관없이 반드시 한 번 실행

## throw문

에러 객체를 생성한다고 에러가 발생하지 않는다.
따라서 throw문을 통해 에러 발생을 유도한다.

## 에러의 전파

에러는 호출자(caller) 방향으로 전파된다.
